src/fetcher.ts

```diff
@@ -9,14 +9,16 @@ import {
   RequestModes,
 } from './request';
 
+export type Fetch = typeof fetch;
 export type Path = string;
 
 export interface IClientConfiguration {
   middleware?: Middleware<any>;
+  fetch?: typeof fetch;
 }
 
-const fetchRequest: Dispatch<Response> = request =>
-  fetch(asNativeRequest(request));
+const fetchRequestFactory = (fetchFn: Fetch) => (request: IHttpRequest) =>
```

Comment:
Let's keep the name from before since we don't need to specify that this is a factory.

Move the default value to the middleware/factory as an independent middlewares would do.

```suggestion
const fetchRequest = (client: Fetch = fetch) => (request: IHttpRequest) =>
```